sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -261,6 +260,14 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
       p.withNewChildren(p.children.map(c => insertCoalesce(c, shouldDisable)))
   }
 
+  /** Inserts a shuffle coalesce after every shuffle */
+  private def insertShuffleCoalesce(plan: SparkPlan): SparkPlan = plan match {
```

Comment:
This one says "after every shuffle", but it really is only after every shuffle, if the RAPIDS shuffle isn't enabled, and the coalesce bit happens on the host, do you think something like this might make it clearer?

```suggestion
  private def insertShuffleCoalesceOnHost(plan: SparkPlan): SparkPlan = plan match {
```

Reply:
Personally I don't want to flag everywhere in the code that `ShuffleCoalesceExec` is really `ShuffleCoalesceOnHost`.  If we feel strongly enough that we need to emphasize the host part of it then let's just change the name of the query plan node.

Reply:
This is definitely in the nit category. The comment says "after every shuffle" but isn't always the case, so a comment tweak may be more appropriate in this scenario.